lib/matplotlib/tests/test_axes.py

```diff
@@ -4452,6 +4452,18 @@ def test_set_get_ticklabels():
     ax[1].set_yticklabels(ax[0].get_yticklabels())
 
 
+def test_subsampled_ticklabels():
+    # test issue 11937
+    fig, ax = plt.subplots()
+    ax.plot(np.arange(10))
+    ax.xaxis.set_ticks(np.arange(10) + 0.1)
+    ax.locator_params(nbins=5)
+    ax.xaxis.set_ticklabels([c for c in "bcdefghijk"])
```

Comment:
Putting these together would make it a bit clearer:
```suggestion
    ax.xaxis.set(ticks=np.arange(10) + 0.1)
                 ticklabels=[c for c in "bcdefghijk"])
    ax.locator_params(nbins=5)
```

Reply:
The use of ``set`` here would work only by accident, because the kwargs are sorted in inverse order, so that set_ticks will be called before set_ticklabels.  I will stick with the individual function calls.

Reply:
This is the closest analogue to `plt.xticks(locs, labels)`, so if you're saying it might be broken, then I think that's not good. Especially given the recent tweet about `.set()`...

Reply:
I don't use twitter and haven't seen the tweet.

I'm not saying the behavior is currently broken, just that it works in this case (I think--I haven't tried it) because the names are sorted in inverse order, and the "s" in "ticks" makes it sort before the first "l" in "ticklabels".  It's an accident.  It might be better to make it explicit by adding a "ticks" entry to Artist._prop_order to boost its priority.  The _prop_order dict provides a way of setting priority at a higher level than the alphanumeric sort.   Presently, it serves only to give 'color' low priority, so it is executed last.

Reply:
[The is the tweet](https://twitter.com/matplotlib/status/1256364300492513281) that @story645 mentioned.

For users, working by accident and working by design are the same thing. So I meant if _this PR_ breaks it, that would not be good. However, it doesn't, so we're fine, even if it's working by accident right now.

Reply:
I guess I'll take advantage of this to point to https://github.com/matplotlib/matplotlib/pull/16328 :)